README.md

```diff
@@ -10,6 +10,15 @@ registry.
 
 *see the [cargo docs](https://doc.rust-lang.org/cargo/reference/registries.html#running-a-registry) for details*
 
+## Dependencies
```

Comment:
i'm inclined to include only dependencies that won't be pulled in automatically. This would include C/C++ headers/system library dependencies (if it had any).

i do think it's worth mentioning the compiler minimum version though!
unstable features are documented here- https://doc.rust-lang.org/unstable-book/index.html
this feature that we're depending on is called `map_first_last`. there's probably a way to find out what the compiler version was when that was merged, and then that becomes the minimum version. need to do some more reading!